cmake/Modules/FindARM.cmake

```diff
@@ -141,19 +141,21 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
       MARK_AS_ADVANCED(${lang}_${type}_FOUND ${lang}_${type}_FLAGS)
     ENDMACRO()
 
-    # Check for SVE256 vector length
+    # Check for different SVE vector lengths
+    CHECK_SVE(CXX "SVE128" "-march=armv8-a+sve -msve-vector-bits=128")
     CHECK_SVE(CXX "SVE256" "-march=armv8-a+sve -msve-vector-bits=256")
+    CHECK_SVE(CXX "SVE512" "-march=armv8-a+sve -msve-vector-bits=512")
 
-    # If SVE256 support is not found, set CXX_SVE_FOUND to FALSE and notify the user
-    if(NOT CXX_SVE256_FOUND)
+    # If SVE128 and SVE256 and SVE512 support is not found, set CXX_SVE_FOUND to FALSE and notify the user
+    if(NOT CXX_SVE128_FOUND AND NOT CXX_SVE256_FOUND AND NOT CXX_SVE512_FOUND)
       set(CXX_SVE_FOUND FALSE CACHE BOOL "SVE not available on host")
       message(STATUS "No SVE processor on this machine.")
```

Comment:
The host machine doesn't have to have SVE support just the toolchain right?

Reply:
Yes, that's right. You just need a toolchain (such as GCC or Clang) that has SVE support. Hardware doesn't need to have SVE support.